### PR TITLE
Add account locking for failed OTP case

### PR DESCRIPTION
**Pull request by nilasini**
## Purpose
Backporting https://github.com/wso2-extensions/identity-outbound-auth-email-otp/pull/86
Support account locking for email-otp when configured as second factor 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: component/authenticator/pom.xml [hunk 1]
```diff
@@ -66,11 +66,15 @@
...
+                            org.wso2.carbon.identity.handler.event.account.lock.service;
+                            version="${carbon.identity.account.lock.handler.imp.pkg.version.range}",
+                            org.wso2.carbon.identity.handler.event.account.lock.exception;
+                            version="${carbon.identity.account.lock.handler.imp.pkg.version.range}",
                             *;resolution:=optional
                         </Import-Package>
                         <Export-Package>
                             !org.wso2.carbon.identity.authenticator.emailotp.internal,
-                            org.wso2.carbon.identity.authenticator.emailotp.*
+                            org.wso2.carbon.identity.authenticator.emailotp.*,
```
Comment (sachiniWettasinghe) [inline component/authenticator/pom.xml:77]:
```suggestion
                            org.wso2.carbon.identity.authenticator.emailotp.*
```

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/internal/EmailOTPServiceDataHolder.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sachiniWettasinghe) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/internal/EmailOTPServiceDataHolder.java]: Add correct licence

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/internal/EmailOTPServiceDataHolder.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ */
+package org.wso2.carbon.identity.authenticator.emailotp.internal;
+
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.handler.event.account.lock.service.AccountLockService;
+
+/**
+ *  Encapsulates the data of EmailOTP authenticator.
+ */
+public class EmailOTPServiceDataHolder {
```
Comment (sachiniWettasinghe) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/internal/EmailOTPServiceDataHolder.java:27]: Add one line below.

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/internal/EmailOTPServiceDataHolder.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    private static EmailOTPServiceDataHolder emailOTPServiceDataHolder;
+    private AccountLockService accountLockService;
+    private IdentityGovernanceService identityGovernanceService;
+
+    public static EmailOTPServiceDataHolder getInstance() {
```
Comment (sachiniWettasinghe) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/internal/EmailOTPServiceDataHolder.java:33]: Add one line below

File: component/authenticator/pom.xml (hunk 1 above)
Comment (nilasini) [reply component/authenticator/pom.xml:77]: without the comma authenticator will not get deployed.

**Review by sachiniWettasinghe — APPROVED** [review]

**Pull request merged**